timeseries/src/autogluon/timeseries/models/sktime/models.py

```diff
@@ -7,11 +9,94 @@
 
 
 class ThetaModel(AbstractSktimeModel):
+    """Theta model for forecasting.
+
+    This is a special case of AutoETS model that can only be applied to positive data.
+
+    See `AbstractSktimeModel` for common parameters.
+
+    Other Parameters
+    ----------------
+    initial_level : float or None, default = None
+        The alpha value of the simple exponential smoothing, if the value is set then
+        this will be used, otherwise it will be estimated from the data.
+    deseasonalize : bool, default = True
+        If True, data is seasonally adjusted.
+    seasonal_period: int or None, default = None
+        Number of time steps in a complete seasonal cycle for seasonal models. For
+        example, 4 for quarterly data with an annual cycle, or 7 for daily data with a
+        weekly cycle.
+        When set to None, seasonal_period will be inferred from the frequency of the
+        training data. Can also be specified manually by providing an integer > 1.
+    fail_if_misoconfigured: bool, default = False
+        If True, the model will raise an exception and fail when given an invalid
+        configuration (e.g., selected seasonality is incompatible with seasonal_period).
+        If False, the model will instead raise a warning and try to adjust the
+        configuration (e.g., turn off seasonality).
+        Setting this parameter to True is useful during HPO to avoid training multiple
+        models that all fall back to the same configuration.
+    """
+
     sktime_forecaster_class = ThetaForecaster
     sktime_allowed_init_args = ["initial_level", "deseasonalize", "sp"]
 
+    def _get_skt_forecaster_args(self, args: dict, min_length: int, inferred_period: int = 1):
+        fail_if_misoconfigured = args.pop("fail_if_misconfigured", False)
+        deseasonalize = args.pop("deseasonalize", True)
+        seasonal_period = args.pop("seasonal_period", None)
+        # Handle the case when seasonal_period is intentionally set to None
+        if seasonal_period is None:
+            seasonal_period = inferred_period
+        args["deseasonalize"] = deseasonalize
+        args["sp"] = seasonal_period
+
+        if deseasonalize and (min_length < 2 * seasonal_period or seasonal_period <= 1):
+            error_message = (
+                f"{self.name} with deseasonalize = {deseasonalize} requires training series of length "
+                f"at least 2 * seasonal_period and seasonal_period > 1 "
+                f"(received min_length = {min_length} and seasonal_period = {seasonal_period})."
+            )
+            if fail_if_misoconfigured:
+                raise ValueError(error_message)
+            else:
+                warnings.warn(error_message + "\nSetting seasonality to None.")
```

Comment:
Is `logging.info` better suited for this type of information or warning is fine?

Reply:
I'd go with 

```
logger = logging.getLoger(__name__)

...

logger.warn(..)
```